### ISSUE TITLE
Access a Modules WebAssembly.Table and WebAssembly.Memory

### ISSUE THREAD

**Issue by EPNW** [emscripten-core/emscripten#14326]
I need to call a function by its index in the `WebAssembly.Table`. How can I access the table of a module?

The same goes for memory, how can I get a reference to `WebAssembly.Memory`?
I know that I can explecitly passs `Module['wasmMemory']` when creating the module and then the emscripten generated js will use this as memory. But I don't want to create the memory on my own since I don't know the settings that were used when the module was compiled. Rather I want the emscripten generated js to create the memory and then access it.
I currently use `Module['HEAP8']` to access the buffer. Is this safe?

**Comment by kripken** [comment]
You can export things on the Module object if you want them. The relevant internal names are wasmTable and wasmMemory, I believe. Note that the memory may only exist after the wasm is compiled, so you may need to add JS to export them using a mechanism like onRuntimeInitialized.

**Comment by EPNW** [comment]
Hm what you prupose would involve modifiyng the generated JavaScript file? I rather not touch it... Is there any build in way to access the table, based on the `Module` object?

**Comment by kripken** [comment]
No, you do not need to modify it after compilation. You can use a `--pre-js` for example to add some JS into the build, that exports more things, https://emscripten.org/docs/tools_reference/emcc.html#emcc-pre-js

**Comment by EPNW** [comment]
So there is no way to achive this without writing JavaScript?

**Comment by sbc100** [comment]
Interesting, it looks like there is currently no way to export `HEAP*`, `wasmMemory`, `wasmTable` via the command line.  

Do you would need as some custom JS as @kripken said.    The reason is that are not consideres runtime methods of native methods and so cannot be passed to `EXPORTED_FUNCTIONS` or `EXPORTED_RUNTIME_METHODS`.

Here is a little example:

```
$ cat pre.js 
Module = {
  getMemory: function () { return wasmMemory }
}
$ cat run.js 
m = require('./a.out.js')
m.onRuntimeInitialized = function() {
  console.log(m.getMemory())
}
$ emcc --pre-js pre.js tests/hello_world.c -O3 && node run.js 
Memory [WebAssembly.Memory] {}
hello, world!
```
